.circleci/scripts/git-diff-develop.ts

```diff
@@ -0,0 +1,94 @@
+import { promisify } from 'util';
+import { exec as execCallback } from 'child_process';
+import fs from 'fs';
+import path from 'path';
+
+const exec = promisify(execCallback);
+
+/**
+ * Fetches the git repository with a specified depth.
+ *
+ * @param {number} depth - The depth to use for the fetch command.
+ * @returns {Promise<boolean>} True if the fetch is successful, otherwise false.
+ */
+async function fetchWithDepth(depth: number): Promise<boolean> {
+  try {
+    await exec(`git fetch --depth ${depth} origin develop`);
+    await exec(`git fetch --depth ${depth} origin ${process.env.CIRCLE_BRANCH}`);
+    return true;
+  } catch (error: any) {
```

Comment:
Nit: This should be disallowed by our lint rules, but our configuration is currently broken and is not applying TypeScript lint rules to this file.

We can avoid using `any` here by using `unknown` instead. TypeScript will then complain about us using `error.message`, but this is easily resolved with a type check (e.g. `error instanceof Error`), or by passing `error` directly into `console.error (e.g. `console.error(`Failed to fetch with depth ${depth}:`, error);`)

Reply:
fixed